rfc/cdevents-translator.md

```diff
@@ -0,0 +1,237 @@
+Call `plugin.Serve` to serve a plugin from the main function to GRPC server using hashicorp's go-plugin
+
+Build the `gerrit-translator-cdevents` plugin using `go build`, so that the plugin can be loaded/used from other Golang main application to `TranslateEvent`
+
+- Example structure of `gerrit-translator-cdevents` plugin:
+````go
+package main
+
+
+type GerritTranslator struct {
+   //Fields for Gerrit event Translator
+}
+
+func (translator *GerritTranslator) TranslateEvent(event string) {
+	//handle translate Gerrit event to CDEvent
+}
+
+func main() {
+	//using hashicorp's plugin
+	plugin.Serve(&plugin.ServeConfig{
+		.....
+
+		GRPCServer: plugin.DefaultGRPCServer,
+	})
+  
+}
+````
+#### GRPC client and server implementations using protocol buffers
+Creating GRPC Client and Server implementations is a common template that can be generated 
+by using [Buf CLI's](https://buf.build/docs/generate/tutorial) `buf generate` command</br>
+That generates Go code from [protocol buffers](https://protobuf.dev/getting-started/gotutorial/) files(.proto) using Go plugin `protoc-gen-go-grpc`.
+
+- Example structure of `cdevents-translator` proto file
+````proto
+// cdevents_translator.proto
+
+syntax = "proto3";
+package proto;
+option go_package = "./proto";
+
+message TranslateRequest {
+  string text = 1;
+}
+
+message TranslateResponse {
+  string result = 1;
+}
+
+message Empty {}
+
+service EventTranslator {
+  rpc TranslateEvent (TranslateRequest) returns (TranslateResponse);
+}
+````
+
+The GRPC client/server will help to handle `EventTranslator` service request/response over GRPC by using the proto generated files.
+
+````go
+package cdeventstranslator
+
+type GRPCClient struct {
+   //init client
+}
+
+func (m *GRPCClient) TranslateEvent(event string) error {
+	_, err := m.client.TranslateEvent(context.Background(), &proto.TranslateRequest{Event: event})
+	return err
+}
+
+type GRPCServer struct {
+	//init server
+}
+
+func (m *GRPCServer) TranslateEvent(ctx context.Context, req *proto.TranslateRequest) (*proto.Empty, error) {
+	return &proto.Empty{}, m.Impl.TranslateEvent(req.Event)
+}
+
+````
+
+The GRPC client/server implementations can be [generated in other languages](https://protobuf.dev/getting-started/), to enable Cross-language support for Plugins implemented.</br>
+Plugin users use `plugin.Client` to launch a subprocess and request an interface implementation using hashicorp's go-plugin over RPC.</br>
+More detailed implementation can be referred from HashiCorp's go-plugin [gRPC examples](https://github.com/hashicorp/go-plugin/tree/main/examples/grpc) 
+
+#### Pros and Cons
+- This structure allows you to have a main application that dynamically loads and communicates with different translator plugins over gRPC.
+- The plugins can be developed independently and reside in separate repositories.</br>
+- HashiCorp's `go-plugin` library developed under Mozilla Public License.
```

Comment:
Is this a pro or a cons? The Mozilla license is recognised by OSI https://opensource.org/license/mpl-2-0/ - do you see an issue with it?